tensorflow-framework/src/main/java/org/tensorflow/framework/activations/Activation.java

```diff
@@ -14,24 +14,104 @@
 =======================================================================*/
 package org.tensorflow.framework.activations;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
 import org.tensorflow.Operand;
 import org.tensorflow.op.Ops;
 import org.tensorflow.types.family.TNumber;
 
-/**
- * Interface for Activations
- *
- * @param <T> the data type of the input and the result
- */
+/** Interface for Activations */
 @FunctionalInterface
-public interface Activation<T extends TNumber> {
+public interface Activation {
+  Map<String, Supplier<Activation>> createMap =
+      new HashMap<String, Supplier<Activation>>() {
+        {
+          put("elu", ELU::new);
+          put("exponential", Exponential::new);
+          put("gelu", GELU::new);
+          put("hard_sigmoid", HardSigmoid::new);
+          put("linear", Linear::new);
+          put("relu", ReLU::new);
+          put("selu", SELU::new);
+          put("sigmoid", Sigmoid::new);
+          put("softmax", Softmax::new);
+          put("softplus", Softplus::new);
+          put("softsign", Softsign::new);
+          put("swish", Swish::new);
+          put("tanh", Tanh::new);
+        }
+      };
+
+  Map<String, Function<Map<String, Object>, Activation>> createMapConfig =
+      new HashMap<String, Function<Map<String, Object>, Activation>>() {
+        {
+          put("elu", ELU::new);
+          put("exponential", Exponential::new);
+          put("gelu", GELU::new);
+          put("hard_sigmoid", HardSigmoid::new);
+          put("linear", Linear::new);
+          put("relu", ReLU::new);
+          put("selu", SELU::new);
+          put("sigmoid", Sigmoid::new);
+          put("softmax", Softmax::new);
+          put("softplus", Softplus::new);
+          put("softsign", Softsign::new);
+          put("swish", Swish::new);
+          put("tanh", Tanh::new);
+        }
+      };
```

Comment:
Why not using Enums instead of Strings? I think it's more idiomatic to Python to use strings everywhere like this, but using Enums in Java would be relieving, if it's not just for the sake of auto-completeness.

We just need to make sure that our Enums are named based on these labels so we can match back a String from Keras config if needed (or we add an attribute to the enums that matches this label).

Reply:
Changed to enums